examples/http/server.cc

```diff
@@ -19,6 +19,15 @@ class RequestHandler : public HTTP_SERVER_NS::HttpRequestCallback
     options.kind          = opentelemetry::trace::SpanKind::kServer;  // server
     std::string span_name = request.uri;
 
+    // extract context from http header
+    const HttpTextMapCarrier<std::map<std::string, std::string>> carrier(
+        (std::map<std::string, std::string> &)request.headers);
+    auto prop = opentelemetry::context::propagation::GlobalTextMapPropagator::GetGlobalPropagator();
+    auto current_ctx = opentelemetry::context::RuntimeContext::GetCurrent();
+    auto new_context = prop->Extract(carrier, current_ctx);
+    options.parent   = GetSpanFromContext(new_context)->GetContext();
```

Comment:
I think this shouldn't be strictly necessary, as `Extract` sets the span as the currently active span in the context?

Reply:
@pyohannes - `Extract` would return a new context with the `span` created from extracted `span_context`, and won't make it active. This is as per the [specs ](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/context/api-propagators.md#textmap-extract):

``
Returns a new Context derived from the Context passed as argument.
``